client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelDefinition.java

```diff
@@ -124,14 +138,69 @@ public Builder setTrainedModel(TrainedModel trainedModel) {
             return this;
         }
 
+        public Builder setInput(Input input) {
+            this.input = input;
+            return this;
+        }
+
         private Builder setTrainedModel(List<TrainedModel> trainedModel) {
             assert trainedModel.size() == 1;
             return setTrainedModel(trainedModel.get(0));
         }
 
         public TrainedModelDefinition build() {
-            return new TrainedModelDefinition(this.trainedModel, this.preProcessors);
+            return new TrainedModelDefinition(this.trainedModel, this.preProcessors, this.input);
+        }
+    }
+
+    public static class Input implements ToXContentObject {
+
+        public static final String NAME = "trained_mode_definition_input";
+        public static final ParseField FEATURE_NAMES = new ParseField("feature_names");
+
+        @SuppressWarnings("unchecked")
+        public static final ConstructingObjectParser<Input, Void> PARSER = new ConstructingObjectParser<>(NAME,
+                true,
+                a -> new Input((List<String>)a[0]));
+        static {
+            PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), FEATURE_NAMES);
+        }
+
+        public static Input fromXContent(XContentParser parser) throws IOException {
+            return PARSER.parse(parser, null);
         }
+
+        private final List<String> featureNames;
+
+        public Input(List<String> featureNames) {
+            this.featureNames = featureNames;
```

Comment:
I think you should either put `requireNonNull` here or add `if` statement in line 186.

Reply:
I don't think this comment got applied.